DOCSP-42762: Indexes (#74)
mongodb/docs-mongoid: source/reference/indexes.txt

named ``cuisine_index`` on the ``cuisine`` field in the ``Restaurant`` class:

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :emphasize-lines: 8, 11
  :start-after: start create index
  :end-before: end create index

The ``index`` macro defines the index you want to create and the ``create_indexes``
macro creates the index in the ``restaurants`` collection. 

When defining an index, the first hash object contains the field you want to
index and its direction. ``1`` represents an ascending index, ``-1`` represents a
descending index. The second hash object contains index options. To learn more
about index options, see the :ref:`mongoid-indexes-api-documentation` section.

Aliases and Declaring Indexes
+++++++++++++++++++++++++++++

You can use aliased field names in index definitions. For example, the following
example creates a field on the ``b`` field, which is an alias of the ``borough``
field:

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :start-after: start create alias index
  :end-before: end create alias index

Create an Index on Embedded Document Fields
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can define an index on embedded document fields. The following code example
> shows how to declare an ascending index on the ``street`` field in the
> ``Restaurant`` class where ``street`` is a field embedded within the ``address``
> field.

Reviewer: [s] This description was hard for me to parse, I wonder if something like this works:
```suggestion
You can define an index on embedded document fields. The following code example
shows how to declare an ascending index on the ``street`` field, which is embedded within the ``address``field in the ``Restaurant`` class.
```

Or break it up:
```suggestion
You can define an index on embedded document fields. The following code example
shows how to declare an ascending index on the ``street`` field in the
``Restaurant`` class. The ``street`` field is embedded within the ``address``
field.
```

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :start-after: start create embedded index
  :end-before: end create embedded index

Create a Compound Index
~~~~~~~~~~~~~~~~~~~~~~~

You can define a compound index on multiple fields. The following code example
shows how to declare a compound index that is ascending on the ``address.street``

> field and descending on the ``name`` field.

Reviewer: [q] This description didn't seem to match the included code example which uses ``borough``. Should it be ``address.street`` in the example instead?

Author: oops, thanks for catching!

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :start-after: start create compound index
  :end-before: end create compound index

Create a Geospatial Index
~~~~~~~~~~~~~~~~~~~~~~~~~

You can define a 2dsphere index on fields that contain GeoJSON objects or
coordinate pairs.
The following example defines a 2dsphere index on a field that contains GeoJSON
objects:

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :start-after: start create 2dsphere index
  :end-before: end create 2dsphere index

> For more information on 2dsphere indexes, see the :manual:`2dsphere </core/indexes/index-types/geospatial/2dsphere/>`

Reviewer: [n] In the deploy preview it looks like there's an extra space before manual.

guide in the {+mdb-server+} manual.

For more information on the GeoJSON type, see the :manual:`GeoJSON Objects </reference/geojson/>`

> guide in the {+mdb-server+} manual.

Reviewer: Oh it looks like this variable isn't in `snooty.toml`. Instead I see `server-manual = "Server manual"` so just need to update here.

Create a Sparse Index
~~~~~~~~~~~~~~~~~~~~~

You can define a sparse index on fields that are not present in all documents. 
The following code example defines a sparse index on the ``borough`` field:

.. literalinclude:: /includes/reference/indexes.rb
  :language: ruby
  :start-after: start create sparse index
  :end-before: end create sparse index

For more information on sparse indexes, see the :manual:`Sparse Indexes </core/index-sparse/>`

> guide in the {+mdb-server+} manual.

Reviewer: Same as above.
